src/Morphir/Visual/ViewValue.elm

```diff
@@ -214,48 +215,52 @@ viewValueByLanguageFeature config value =
                                 [ viewValue config subjectValue
-                                , el [ Font.bold ] (text ".")
-                                , text (toCamelCase fieldName)
+                                , el [Font.bold] <| text "→"
+                                , readableFieldName
                                 ]
                     in
-                    case Config.evaluate (subjectValue |> Value.toRawValue) config of
-                        Ok valueType ->
-                            case valueType |> fromRawValue config.ir of
-                                Ok (Value.Variable ( index, _ ) variableName) ->
-                                    let
-                                        variableValue : Maybe RawValue
-                                        variableValue =
-                                            Dict.get variableName config.state.variables
-                                    in
-                                    el
-                                        [ onMouseEnter (config.handlers.onHoverOver index config.nodePath variableValue)
-                                        , onMouseLeave (config.handlers.onHoverLeave index config.nodePath)
-                                        , Element.below
-                                            (if config.state.popupVariables.variableIndex == index then
-                                                el [ smallPadding config.state.theme |> padding ] (viewPopup config)
-
-                                             else
-                                                Element.text "Not Found"
-                                            )
-                                        , center
-                                        ]
-                                        (String.concat
-                                            [ "the "
-                                            , nameToText variableName
-                                            , "'s "
-                                            , nameToText fieldName
-                                            ]
-                                            |> text
-                                        )
-
-                                _ ->
-                                    defaultValue
+                    case subjectValue of
+                        Value.Variable ( index, _ ) variableName ->
+                            let
+                                variableValue : Maybe RawValue
+                                variableValue =
+                                    Dict.get variableName config.state.variables
+
+                                singularOrPlural vname =
+                                    if ((vname |> List.Extra.last |> Maybe.map (String.toList >> List.Extra.last >> Maybe.withDefault '_')) |> Maybe.withDefault '_') == 's' then
+                                        "' "
```

Comment:
This is very difficult to read. Can you move out getting the last letter of the last word into its own function within this same let?